sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/OptimizeSkewInRebalancePartitions.scala

```diff
@@ -92,9 +93,21 @@ object OptimizeSkewInRebalancePartitions extends AQEShuffleReadRule {
       return plan
     }
 
-    plan transformUp {
+    val optimized = plan transformUp {
       case stage: ShuffleQueryStageExec if isSupported(stage.shuffle) =>
```

Comment:
since we move the rule to prepare execution, we should make sure the stage is materialized and fast return if it's not materialized.